plugin/externalPluginEntry.go

```diff
@@ -516,12 +516,40 @@ func newStdoutDecodeErr(ctx context.Context, decodedThing string, reason error,
 	return newInvokeError(fmt.Sprintf("could not decode %v from stdout: %v", decodedThing, reason), inv)
 }
 
+func (e *externalPluginEntry) pluginName() string {
+	pluginName, _ := e.splitTypeID()
+	return pluginName
+}
+
+func (e *externalPluginEntry) rawTypeID() string {
+	_, rawTypeID := e.splitTypeID()
+	return rawTypeID
+}
+
+// splits the type ID into a tuple (<plugin_name>, <raw_type_id>)
+func (e *externalPluginEntry) splitTypeID() (string, string) {
+	substrings := strings.SplitN(e.TypeID(), "::", 2)
+	if len(substrings) < 2 {
+		return "__unknown__", ""
```

Comment:
Should the 2nd return `substrings[0]`? I assume SplitN always returns at least one, for an empty string.

Reply:
Hmm, that case would correspond to some entries specifying type IDs and some entries not specifying type IDs, which is impossible in the (common) entry schema scenario. I figured if we hit that case, then the type ID doesn't make sense. Returning `substrings[0]` might not make sense either in case the author namespaces their type IDs using `::`, but maybe we could document something along the lines of "If you're using type IDs w/o entry schema support, then don't namespace them with '::' else things will go wrong".

I don't have strong feelings about either approach.

Reply:
I think it's worth commenting how we might get here and it's not an error.